SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialFastArrayNetSerialize.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EngineClasses/SpatialFastArrayNetSerialize.h"
+
+#include "EngineClasses/SpatialNetBitReader.h"
+#include "EngineClasses/SpatialNetBitWriter.h"
+
+namespace improbable
+{
+
+FSpatialNetDeltaSerializeInfo FSpatialNetDeltaSerializeInfo::CreateWriter(FSpatialNetBitWriter& Writer, SpatialFastArrayNetSerializeCB& Callback)
+{
+	FSpatialNetDeltaSerializeInfo NetDeltaInfo;
+
+	NetDeltaInfo.Writer = &Writer;
+	NetDeltaInfo.Map = Writer.PackageMap;
+	NetDeltaInfo.NetSerializeCB = &Callback;
+
+	return NetDeltaInfo;
+}
+
+FSpatialNetDeltaSerializeInfo FSpatialNetDeltaSerializeInfo::CreateReader(FSpatialNetBitReader& Reader, SpatialFastArrayNetSerializeCB& Callback)
+{
+	FSpatialNetDeltaSerializeInfo NetDeltaInfo;
+
+	NetDeltaInfo.Reader = &Reader;
+	NetDeltaInfo.Map = Reader.PackageMap;
+	NetDeltaInfo.NetSerializeCB = &Callback;
+
+	return NetDeltaInfo;
+}
+
+void SpatialFastArrayNetSerializeCB::NetSerializeStruct(UScriptStruct* Struct, FBitArchive& Ar, UPackageMap* PackageMap, void* Data, bool& bHasUnmapped)
+{
+	if (Struct->StructFlags & STRUCT_NetSerializeNative)
```

Comment:
Can you comment about what this function is doing? If the struct isn't NetSerializeNative then what is it instead?

Reply:
This is doing a network serialize on a struct. That struct doesn't necessarily have the NetSerialize trait though, and if it doesn't will just do a normal rep layout serialize.